src/bonsai/bonsai/bim/module/sequence/operator.py

```diff
@@ -654,6 +654,75 @@ def execute(self, context):
         return {"FINISHED"}
 
 
+class ImportAlignmentCSV(bpy.types.Operator, tool.Ifc.Operator, ImportHelper):
+    bl_idname = "bim.import_alignment_csv"
+    bl_label = "Import Alignment CSV"
+    bl_options = {"REGISTER", "UNDO"}
+    filename_ext = ".csv"
+    filter_glob: bpy.props.StringProperty(default="*.csv", options={"HIDDEN"})
+
+    @classmethod
+    def poll(cls, context):
+        ifc_file = tool.Ifc.get()
+        if ifc_file is None:
+            cls.poll_message_set("No IFC file is loaded.")
+            return False
+        return True
+
+    def _execute(self, context):
+        import ifcopenshell.api.alignment
+        self.file = tool.Ifc.get()
+        start = time.time()
+        alignment = ifcopenshell.api.alignment.create_alignment_from_csv(self.file,self.filepath)
+
+        # IFC 4.1.5.1 alignments cannot be contained in spatial structures, but can be referenced into them
+        sites = self.file.by_type("IfcSite")
+        for site in sites:
+            ifcopenshell.api.spatial.reference_structure(self.file,products=[alignment],relating_structure=site)
+
+        # process the generated IfcReferent for the alignment
+        for rel in alignment.IsNestedBy:
+            for referent in rel.RelatedObjects:
+                if referent.is_a().upper() == "IFCREFERENT":
```

Comment:
is_a("IfcReference")

Also recommend using util.element instead of manually going through rels.